### ISSUE TITLE
calculating loss while finetuning

### ISSUE THREAD

**Issue by RaiAmanRai** [antecessor/mpt_7b_fine_tuning#1]
hi, thnanks for sharing the notebook.

I see that while calculating loss for finetuned examples, you only calculate the loss for the first logit generated and the first tokenized logit for the example. I might be wrong here, but why not do it for the complete sequence generated and the input example sequence?

after a epoch the model would learn the first logit but then it keeps repeating sentences.
whats your suggestion on this?
